### DISCUSSION TITLE
Device Reachability Status - Is it mandatory to support all values?

### DISCUSSION THREAD

**Discussion by MsOlaAjibola** [camaraproject/DeviceStatus#227] [Q&A]
Currently there are 3 reachability status values - CONNECTED_SMS, CONNECTED_DATA and NOT_CONNECTED. Is it mandatory for an Operator to support all 3? For example could an Operator choose to not support CONNECTED_SMS?

Thanks.

**Comment by eric-murray** [comment]
Hi @MsOlaAjibola

It depends on the use case the API is being used for.

If a mobile device was connected to your network, but the user of that device had disabled mobile data, then what would your API return? If you did not support `CONNECTED_SMS`, then you could either:
- return `CONNECTED_DATA` to indicate that the device was connected the network; or
- return `NOT_CONNECTED` to indicate that the device does not have data connectivity?

Neither response is entirely correct, but may be suitable for the use case intended by the API consumer.

In practise, checking whether the device has packet switched and/or circuit switched connectivity is not so difficult.

**Comment by bigludo7** [comment]
@MsOlaAjibola @eric-murray : As I do not see impact on code could we transfer this issue in Discussion?

**Comment by MsOlaAjibola** [comment]
Hi @bigludo7 , 

Yes, let's do that. Thanks
